public/js/admin.js

```diff
@@ -0,0 +1,306 @@
+        var url = "/dashboard/channel/create";
+        var token               = $("#token").val();
+        var channel_name        = $("#name").val();
+        var channel_description = $("#description").val();
+        var data =
+            {
+                url        : url,
+                parameter  :
+                {
+                    _token        : token,
+                    name          : channel_name,
+                    description   : channel_description
+                }
+            }
+        processAjax(data.url, data.parameter, data.parameter.channel_name );
+        return false;
+    });
+
+    /**
+     * onSubmit event to handle Channel update
+     */
+    $("#channel_update").submit( function () {
+        var url                 = "/dashboard/channel/edit";
+        var token               = $("#token").val();
+        var channel_id          = $("#channel_id").val();
+        var channel_name        = $("#channel_name").val();
+        var channel_description = $("#channel_description").val();
+        var data =
+            {
+                url        : url,
+                parameter  :
+                {
+                    _token                : token,
+                    channel_id            : channel_id,
+                    channel_name          : channel_name,
+                    channel_description   : channel_description
+                }
+            }
+        processAjax(data.url, data.parameter, data.parameter.channel_name );
+        return false;
+    });
+
+    /**
+     * onSubmit event to handle User Edit
+     */
+    $("#create_user").submit( function () {
+        var url       = "/dashboard/user/create";
+        var token     = $("#_token").val();
+        var username  = $("#username").val();
+        var user_role = $("#user_role").val();
+        var data =
+            {
+                url        : url,
+                parameter  :
+                {
+                    _token      : token,
+                    username    : username,
+                    user_role   : user_role
+                }
+            }
+        processAjax(data.url, data.parameter, data.parameter.user_name );
+        return false;
+    });
+
+    /**
+     * onSubmit event to handle User Edit
+     */
+    $("#edit_user").submit( function () {
+        var url       = "/dashboard/user/edit";
+        var token     = $("#_token").val();
+        var user_id   = $("#user_id").val();
+        var username  = $("#username").val();
+        var user_role = $("#user_role").val();
+        var data =
+            {
+                url        : url,
+                parameter  :
+                {
+                    _token      : token,
+                    user_id     : user_id,
+                    username    : username,
+                    user_role   : user_role
+                }
+            }
+        processAjax(data.url, data.parameter, data.parameter.user_name );
```

Comment:
You used username in the parameter object, but on line 111, you are accessing it via `data.parameter.user_name`. Is that correct?